tests/test_all_rules.py

```diff
@@ -547,6 +547,34 @@ def test_non_bbr_in_correct_dir(self):
                 self.assertEqual(rule.contents.data.building_block_type, None,
                                  f'{self.rule_str(rule)} should be in {proper_directory}')
 
+    def test_rule_change_has_updated_date(self):
+        """Test to ensure modified rules have updated_date field updated."""
+
+        rules_path = get_path("rules")
+        rules__bbr_path = get_path("rules_building_block")
+
+        # Use git diff to check if the file(s) has been modified in rules/ directory
+        # For now this checks even rules/_deprecated but we may combine
+        # with a test case for deprecated rules modification
+        detection_rules_git = make_git()
+        result = detection_rules_git("diff", "--diff-filter=M", "origin/main", "--name-only",
+                                     rules_path, rules__bbr_path)
```

Comment:
I like this idea and where you're going in general with this. Especially since it doesn't require GitHub tokens to run the test.

Just keep in mind that if we want to duplicate this in our other rule repos, they probably dont have `detection_rules_git`. You may be able to explore using the `pygithub` package since we have that as a dependency. If you went that route, it would limit us a bit where the token would have to be set locally before running the test.

Reply:
Agreed, will keep this mind.